builder/builder-next/controller.go

```diff
@@ -90,6 +91,14 @@ func newSnapshotterController(ctx context.Context, rt http.RoundTripper, opt Opt
 	nc := netproviders.Opt{
 		Mode: "host",
 	}
+
+	// HACK! Windows doesn't have 'host' mode networking.
+	if runtime.GOOS == "windows" {
+		nc = netproviders.Opt{
+			Mode: "auto",
+		}
+	}
+
```

Comment:
Side-thought: I wonder if we can teach BuildKit to do host networking on Windows? I don't know off-hand why that's not supported now, as it happens. Maybe a future research project.

Reply:
Oh duh. It's setting the network config for containerised workers, and we can't support that on Windows, according to https://github.com/moby/buildkit/pull/1585. Good on that handsome and intelligent person for answering my question years before I asked it...

So yeah, for now there's no real choice here, it's not currently an active codepath anyway, and maybe someday someone will implement BuildKit not-sandboxed+host-networking on Windows using Host Process Containers.